server/src/main/java/org/elasticsearch/search/aggregations/metrics/min/MinAggregatorFactory.java

```diff
@@ -43,12 +55,100 @@ public MinAggregatorFactory(String name, ValuesSourceConfig<Numeric> config, Sea
     @Override
     protected Aggregator createUnmapped(Aggregator parent, List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData)
             throws IOException {
-        return new MinAggregator(name, null, config.format(), context, parent, pipelineAggregators, metaData);
+        return new MinAggregator(name, null, config.format(), context, parent, pipelineAggregators, metaData, createEmpty());
     }
 
     @Override
     protected Aggregator doCreateInternal(ValuesSource.Numeric valuesSource, Aggregator parent, boolean collectsFromSingleBucket,
             List<PipelineAggregator> pipelineAggregators, Map<String, Object> metaData) throws IOException {
-        return new MinAggregator(name, valuesSource, config.format(), context, parent, pipelineAggregators, metaData);
+        final Function<byte[], Number> converter = getPointReaderOrNull(context, getParent(), config);
+        final CheckedFunction<LeafReader, Number, IOException> shortcutMinFunc;
+        if (converter == null) {
+            shortcutMinFunc = createEmpty();
+        } else {
+            final String fieldName = config.fieldContext().field();
+            shortcutMinFunc = createShortcutMin(fieldName, converter);
+        }
+        return new MinAggregator(name, valuesSource, config.format(), context, parent, pipelineAggregators, metaData, shortcutMinFunc);
+    }
+
+    public static CheckedFunction<LeafReader, Number, IOException> createEmpty() {
+        return (c) -> null;
+    }
+
+    /**
+     * Returns a converter for point values if early termination is applicable to
+     * the context or null otherwise.
+     *
+     * @param context The {@link SearchContext} of the aggregation.
+     * @param parent The parent aggregator factory.
+     * @param config The config for the values source metric.
+     */
+    public static Function<byte[], Number> getPointReaderOrNull(SearchContext context, AggregatorFactory<?> parent,
+                                                                ValuesSourceConfig<ValuesSource.Numeric> config) {
+        if (context.query() != null &&
+                context.query().getClass() != MatchAllDocsQuery.class) {
+            return null;
+        }
+        if (parent != null) {
+            return null;
+        }
+        if (config.fieldContext() != null) {
+            MappedFieldType fieldType = config.fieldContext().fieldType();
+            if (fieldType == null || fieldType.indexOptions() == IndexOptions.NONE) {
+                return null;
+            }
+            Function<byte[], Number> converter = null;
```

Comment:
should we also verify that the script is null?

Reply:
yes good catch